website/source/docs/provisioners/inspec.html.md

```diff
@@ -0,0 +1,135 @@
+---
+description: |
+    The inspec Packer provisioner allows inspec profiles to be run to test the
+    machine.
+layout: docs
+page_title: 'InSpec - Provisioners'
+sidebar_current: 'docs-provisioners-inspec'
+---
+
+# InSpec Provisioner
+
+Type: `inspec`
+
+The `inspec` Packer provisioner runs InSpec profiles. It dynamically creates a
+target configured to use SSH, runs an SSH server, executes `inspec exec`, and
+marshals InSpec tests through the SSH server to the machine being provisioned
+by Packer.
+
+## Basic Example
+
+This is a fully functional template that will test an image on DigitalOcean.
+Replace the mock `api_token` value with your own.
+
+``` json
+{
+  "provisioners": [
+    {
+      "type": "inspec",
+      "profile": "https://github.com/dev-sec/linux-baseline"
+    }
+  ],
+
+  "builders": [
+    {
+      "type": "digitalocean",
+      "api_token": "6a561151587389c7cf8faa2d83e94150a4202da0e2bad34dd2bf236018ffaeeb",
+      "image": "ubuntu-14-04-x64",
+      "region": "sfo1"
+    }
+  ]
+}
+```
+
+## Configuration Reference
+
+Required Parameters:
+
+-   `profile` - The profile to be executed by InSpec.
+
+Optional Parameters:
+
+-   `inspec_env_vars` (array of strings) - Environment variables to set before
+    running InSpec. Usage example:
+
+    ``` json
+    {
+      "inspec_env_vars": [ "AWS_SECRET_KEY_ID=lq9WOb5ME5WxOg5wsIpBz87kh4B8GbZNwd/PDkiq", "AWS_ACCESS_KEY_ID=AKIAI2SN2YBIPZJUW56Q" ]
+    }
+    ```
+
+-   `command` (string) - The command to invoke inspec. Defaults to `inspec`.
+
+-   `extra_arguments` (array of strings) - Extra arguments to pass to InSpec.
+    These arguments *will not* be passed through a shell and arguments should
+    not be quoted. Usage example:
+
+    ``` json
+    {
+      "extra_arguments": [ "--sudo", "--reporter", "json" ]
+    }
+    ```
+
+-   `attributes` (array of strings) - Attribute Files used by InSpec which will
+    be passed to the `--attrs` argument of the `inspec` command when this
+    provisioner runs inspec. Specify this if you want a different location.
+    Note using also `"--attrs"` in `extra_arguments` will override this
+    setting.
+
+-   `attributes_directory` (string) - The directory in which to place the
+    temporary generated InSpec Attributes file. By default, this is the
+    system-specific temporary file location. The fully-qualified name of this
+    temporary file will be passed to the `--attrs` argument of the `inspec`
+    command when this provisioner runs inspec. Specify this if you want a
+    different location.
+
+-   `backend` (string) - Backend used by InSpec for connetction. Defaults to
+    SSH.
```

Comment:
no tested atm. the docker target does not work since the provisioner does not use the container id
maybe i get arround to build a test case for this one and implement it.

Reply:
I've a working example for the docker target.
For the WinRM target, some implementation might be needed to make it more convenient, but i think all parameters can be provided via `extra_arguments`.

https://gist.github.com/xinau/c3839a400fc48f34ccb9a71613aaee56